adapters/repos/db/lsmkv/segment_group_compaction.go

```diff
@@ -24,75 +24,78 @@ import (
 	"github.com/weaviate/weaviate/entities/cyclemanager"
 )
 
-func (sg *SegmentGroup) eligibleForCompaction() bool {
+func (sg *SegmentGroup) bestCompactionCandidatePair() []int {
 	sg.maintenanceLock.RLock()
 	defer sg.maintenanceLock.RUnlock()
 
-	// if true, the parent shard has indicated that it has
-	// entered an immutable state. During this time, the
-	// SegmentGroup should refrain from flushing until its
-	// shard indicates otherwise
-	if sg.isReadyOnly() {
-		return false
-	}
-
-	// if there are at least two segments of the same level a regular compaction
-	// can be performed
-
-	levels := map[uint16]int{}
-
-	for _, segment := range sg.segments {
-		levels[segment.level]++
-		if levels[segment.level] > 1 {
-			return true
-		}
+	// Nothing to compact
+	if len(sg.segments) < 2 {
+		return nil
 	}
 
-	return false
-}
-
-func (sg *SegmentGroup) bestCompactionCandidatePair() []int {
-	sg.maintenanceLock.RLock()
-	defer sg.maintenanceLock.RUnlock()
-
 	// first determine the lowest level with candidates
 	levels := map[uint16]int{}
+	lowestPairLevel := uint16(math.MaxUint16)
+	lowestLevel := uint16(math.MaxUint16)
+	lowestIndex := -1
+	// secondLowestLevel := uint16(math.MaxUint16)
+	secondLowestIndex := -1
+	highestLevel := uint16(0)
+	pairExists := false
+	// var lowestSegment *segment
+	// var secondLowestSegment *segment
```

Comment:
Can these commented variables be removed?

Reply:
done